tests/class-wp-test-rest-controller-testcase.php

```diff
@@ -39,4 +39,8 @@ public function tearDown() {
 
 	abstract public function test_get_item_schema();
 
+	public function test_rest_url_for_leading_slash( $url, $path ) {
+		// Make sure path for rest_url has a leading slash for proper resolution.
+		$this->assertTrue( 0 === strpos( $path, '/' ) );
```

Comment:
Because you're running this test on a filter, you need to make sure to return the `$url` :) Otherwise, the sky will fall upon your head, as it has done.